### ISSUE TITLE
[Help] An error occurs when more objects are created than the number of objects what I can see.

### ISSUE THREAD

**Issue by hyunwoo9120** [qiankanglai/LoopScrollRect#153]
The scroll doesn't work properly and the next object doesn't appear.
(I can see only two objects on the screen, but four objects are created, and when I scroll, all objects in the content are inactive (active: false).)

I use vertical layout group, and it has a padding / spacing value.
![image](https://user-images.githubusercontent.com/53012696/233314536-77e2fd15-7afc-47ea-9c7f-cb5c754c3c99.png)

* Reproduction Step (maybe)
  * open your DemoScene (I only use a VerticalScroll at this scene)
  * set the ScrollCell1 size (like width: 300, height: 200)
  * uncheck the Control Child Size width and height both (Content of Vertical Scroll)
  *  Play and Just Scroll then it doesn't properly

I also unchecked the Content Size Fitter to solve this problem.

**Comment by hyunwoo9120** [comment]
I guess the scroll loop rect calculates content's height inaccurately.

**Comment by qiankanglai** [comment]
LoopScrollRect relies on calculating item size with `LayoutUtility.GetPreferredHeight` or `LayoutUtility.GetPreferredWidth`. So after applying the reproducing steps above, the cell height gets 200 here which is correct.

The problem in fact is cell has zero height after disabling `Control Child Size`
![](https://user-images.githubusercontent.com/1546130/233659900-16c95072-2047-4053-92a0-90795b99392f.png)

You can use other ways to adjust the real cell size (eg. setting cell prefab's height directly) and it just works.

**Comment by hyunwoo9120** [comment]
First, thank your for your answer but I can solve my problem.

set the ScrollCell1 size (like width: 300, height: 200)
-> It means I set the prefab ScrollCell1 size 300x200.

![image](https://user-images.githubusercontent.com/53012696/233878480-8036bba7-d18c-4188-94a3-2294ae824b7e.png)

And I just unchecked the layout element of the prefab because I want to set the size directly.

https://youtu.be/N5R2qKKeHJI
If you can, please watch this video.

**Comment by hyunwoo9120** [comment]
Hi, I found something.

The probelm is happened because the itemTypeEnd is set wrong number. This wrong itemTypeEnd is increased as NewItemAtEnd is called.
Also I found NewItemAtEnd is called more, because sizeToFill and sizedFill (either or both I don't know) is wrong.

.... I hope this comment will help you solve this.

**Comment by hyunwoo9120** [comment]
Oh god I spent lots of time 😂 I finally solved.
If I bother you I am sorry. I'd been focusing on solving this problem for over a week.

Even though it's written in the example,
I thought that if I use my prefab's size (wxh) directly, I don't need the layout element of prefab.

But when I got the item size by getPreferredHeight at GetSize, it returned incorrect number.
So I added the layout element to my prefab and checked preferred width and height, solved this...😂
